subprojects/docs/src/docs/userguide/testKit.adoc

```diff
@@ -72,6 +74,62 @@ The following demonstrates the usage of Gradle runner in a Groovy Spock test:
 
 It is a common practice to implement any custom build logic (like plugins and task types) that is more complex in nature as external classes in a standalone project. The main driver behind this approach is bundle the compiled code into a JAR file, publish it to a binary repository and reuse it across various projects.
 
+[[sec:injecting_gradle_runner_into_spock_test]]
+=== Injecting Gradle runner into Spock tests
+
+When using TestKit with Spock, any field or parameter of type api:org.gradle.testkit.runner.GradleRunner[] that is not manually assigned gets automatically an individual instance injected, each with their own automatically generated temporary project directory if you don't specify an explicit project directory using the `@ProjectDirProvider` annotation on the field or parameter. If an automatically generated temporary project directory is used, it gets deleted when the injected instance gets out of scope.
+
+Shared fields are filled in and cleaned up in a specification interceptor. This means that all iterations of all feature methods see the same value with the same underlying data on disk and also all setup and cleanup methods see those values. They are created before the specification starts and are cleaned up after the specification ends.
+
+Non-shared fields are filled in and cleaned up in an iteration interceptor. This means that the fields are filled in at the start of each iteration. Iteration means that for data-driven feature methods with a `where:` block the fields are filled at the start of each individual iteration and at the start means before any setup methods. The filled in objects and their underlying data on disk is available until the end of the iteration, i. e. until after all cleanup methods.
+
+++++
+<sample id="testKitInjectGradleRunnerIntoFieldsWithSpock" dir="testKit/gradleRunner/gradleRunnerInjectionIntoSpockTests/src/test/groovy/org/gradle/sample" title="Inject GradleRunner into fields with Spock">
+    <sourcefile file="BuildLogicFunctionalTest.groovy" snippet="fields"/>
+</sample>
+++++
+
+Parameters are filled in and cleaned up in a feature method interceptor. This means that each call of the feature method (i. e. each individual iteration) gets individual injected values with individual underlying data on disk and they are cleaned up as soon as the feature method is finished. Those values and their underlying data on disk are not available during setup methods or in cleanup methods.
```

Comment:
Can you please explain this a little bit? I think non-shared instance field can cover this scenario (unless you want two `GradleRunner` in a single invocation,  which is rare IMO).

Reply:
I'm not sure what you mean.
It supports both, instance fields and paramters.
If you e. g. have one test specification with 50 test methods, 47 do unit tests, 3 of them integration tests using the `GradleRunner`, you might not want to waste the time of preparing the `GradleRunner` for each of those 50 tests, but only for the 3 that actually need it.
Of course you could also separate the tests in two specifications, but I don't like forcing people to do it when there is a better way.

Having multiple `GradleRunner` instances has nothing to do with this.
You can have multiple independent `GradleRunner`s by specifying mutliple fields, or multiple parameters or by mixing both.
Each individual field or parameter gets its individual `GradleRunner`.

I agree, that it is probably unlikely that someone needs multiple `GradleRunner` instances within the same test.
But who knows, if he needs it, he can get it.